toonz/sources/toonz/startuppopup.cpp

```diff
@@ -804,10 +818,21 @@ void StartupPopup::onRecentSceneClicked(int index) {
   if (index < 0) return;
   QString path =
       RecentFiles::instance()->getFilePath(index, RecentFiles::Scene);
-  IoCmd::loadScene(TFilePath(path.toStdWString()), false);
-  RecentFiles::instance()->moveFilePath(index, 0, RecentFiles::Scene);
-  RecentFiles::instance()->refreshRecentFilesMenu(RecentFiles::Scene);
-  hide();
+  if (!TSystem::doesExistFileOrLevel(TFilePath(path.toStdWString()))) {
+    RecentFiles::instance()->removeFilePath(index, RecentFiles::Scene);
+    RecentFiles::instance()->refreshRecentFilesMenu(RecentFiles::Scene);
+    for (int i = 0; i < m_recentNamesLabels.count() && i < 7; i++) {
```

Comment:
Oh, please don't forget to fix here as well! 
IMO it would better to define some const value instead of using magic number of 10.